lte/gateway/c/core/oai/tasks/ngap/ngap_amf_nas_procedures.c

```diff
@@ -1201,6 +1201,19 @@ int ngap_amf_nas_pdusession_resource_setup_stream(
 
     ngap_pdusession_setup_item_ies->pDUSessionID = session_item->Pdu_Session_ID;
 
+    ngap_pdusession_setup_item_ies->pDUSessionNAS_PDU =
+        calloc(1, sizeof(Ngap_NAS_PDU_t));
+    ngap_pdusession_setup_item_ies->pDUSessionNAS_PDU->size =
+        blength(pdusession_resource_setup_req->nas_pdu);
+
+    ngap_pdusession_setup_item_ies->pDUSessionNAS_PDU->buf = calloc(
+        blength(pdusession_resource_setup_req->nas_pdu), sizeof(uint8_t));
+
+    memcpy(
+        ngap_pdusession_setup_item_ies->pDUSessionNAS_PDU->buf,
+        bdata(pdusession_resource_setup_req->nas_pdu),
+        ngap_pdusession_setup_item_ies->pDUSessionNAS_PDU->size);
+
```

Comment:
where are these dynamically allocated memory got freed up?

Reply:
IEs allocated dynamically and added to Ngap_NGAP_PDU_t(pdu). 
all IEs memory cleared inside ngap_amf_encode_pdu API by ASN_STRUCT_FREE_CONTENTS_ONLY.

we have a UT case, it was validated memory leak(pdu_sess_resource_setup_req_sunny_day).
Please find latest  NGAP UT results, no leak reported.
![image](https://user-images.githubusercontent.com/83060027/140014426-57ac2d72-b667-44e0-b55a-f8876d8be7a9.png)